src/nunavut/lang/docgen/__init__.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Copyright (C) 2018-2019  UAVCAN Development Team  <uavcan.org>
+# This software is distributed under the terms of the MIT License.
+#
+"""
+    Filters for generating docs. All filters in this
+    module will be available in the template's global namespace as ``docgen``.
+"""
+
+import string
+import random
+import typing
+
+import pydsdl
+
+from ...templates import template_language_filter
+from .. import Language
+
+
+@template_language_filter(__name__)
+def filter_is_composite(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.CompositeType)
+
+
+@template_language_filter(__name__)
+def filter_is_array(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.ArrayType)
+
+
+@template_language_filter(__name__)
+def filter_is_service(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.ServiceType)
+
+
+@template_language_filter(__name__)
+def filter_is_service_req(language: Language, instance: typing.Any) -> bool:
+    return instance.has_parent_service and instance.full_name.split(".")[-1] == "Request"
+
+
+@template_language_filter(__name__)
+def filter_is_field(language: Language, instance: typing.Any) -> bool:
+    return isinstance(instance, pydsdl.Field)
+
+
+@template_language_filter(__name__)
+def filter_extent(language: Language, instance: typing.Any) -> str:
+    try:
+        return instance.bit_length_set.max or 0
```

Comment:
Extent and the maximum element of the bit length set are not the same thing: https://pydsdl.readthedocs.io/en/stable/pages/pydsdl.html#pydsdl._serializable._composite.DelimitedType

Maybe we actually should display both?

- Extent is useful if you need to determine how much memory is needed to deserialize a data type or how much additional data you can add in its next minor version.
- Max length is useful if you need to determine the worst case network bandwidth utilization by the data type.